src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+        for (String name : names) {
+            int totalEarnings = 0;
+            for (String record : data) {
+                String[] parts = record.split(" ");
+                String employeeName = parts[1];
+                if (employeeName.equals(name)) {
+                    if (isDateInRange(parts[0], dateFrom, dateTo)) {
+                        totalEarnings += Integer.parseInt(parts[2]) * Integer.parseInt(parts[3]);
```

Comment:
use constants instead of numbers

Reply:
not fixed, indexes 0 1 2 etc should be constants with meaningful names like NAME_INDEX